ONSdigital/ras-frontstage
# Ras account locked

# Motivation and Context
When the user lock is account because too many attempts, needs to be redirect to a erro page and sent a notification email. 

# What has changed
The sign_in page has been re-factored , the multi if option for oauth2_token to a runtime dictionary that contains all the available defined errors kid encapsulated in classes that all extend RoutingValidation class. In this way the Open Close principle is respected and for each new error or exception is enough to add a new class that describe the error validation behavior. 
New file created : 

1.  sign-in.account-locked.html for the error page  
2. error_routing.py  where all the routing erro classes are created. 
3. notifications.py where the logic for the email gateway service is implemented. 
4. test_sign_in.py added unit tests . 
5. new email in the gov notify server 


 

# How to test?
Create a new user and try to log in with the wrong password for 10 times. The 11th time the user should be redirect to the defined error page and email should be sent ( received ) . 

# Links
https://trello.com/c/6IEaOnA9/269-us204-ext-user-account-locked-l
